private/pkg/git/gitbranch/gitbranch.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitbranch
+
+import (
+	"errors"
+
+	"github.com/bufbuild/buf/private/pkg/git/gitobject"
+)
+
+// Ranger ranges over branches and commits for a git repository.
+//
+// Only branches and commits from the remote named `origin` are ranged.
+type Ranger interface {
```

Comment:
Would call it `BranchIterator`

Reply:
It's also iterating over commits. Do you recommend I split the type?

Reply:
yea sgtm